go/vt/vtgate/vcursor_impl.go

```diff
@@ -230,7 +233,36 @@ func (vc *vcursorImpl) Environment() *vtenv.Environment {
 }
 
 func (vc *vcursorImpl) TimeZone() *time.Location {
-	return vc.safeSession.TimeZone()
+	zone := vc.safeSession.TimeZone()
+	if zone == "" {
+		return nil
+	}
+
+	ast, err := vc.Environment().Parser().ParseExpr(zone)
```

Comment:
Can this really be any arbitrary expression? 

Even if it can, would it not be a lot faster to first try to parse it as a timezone and only if that fails try to evaluate the expression?